read-fonts/src/tables/postscript/dict.rs

```diff
@@ -0,0 +1,718 @@
+        use Operator::*;
+        Some(match opcode {
+            // Top DICT operators
+            0 => Version,
+            1 => Notice,
+            2 => FullName,
+            3 => FamilyName,
+            4 => Weight,
+            5 => FontBbox,
+            13 => UniqueId,
+            14 => Xuid,
+            15 => Charset,
+            16 => Encoding,
+            17 => CharstringsOffset,
+            18 => PrivateDictRange,
+            24 => VariationStoreOffset,
+            // Private DICT operators
+            6 => BlueValues,
+            7 => OtherBlues,
+            8 => FamilyBlues,
+            9 => FamilyOtherBlues,
+            10 => StdHw,
+            11 => StdVw,
+            19 => SubrsOffset,
+            20 => DefaultWidthX,
+            21 => NominalWidthX,
+            22 => VariationStoreIndex,
+            23 => Blend,
+            // Font DICT only uses PrivateDictRange
+            _ => return None,
+        })
+    }
+
+    fn from_extended_opcode(opcode: u8) -> Option<Self> {
+        use Operator::*;
+        Some(match opcode {
+            // Top DICT operators
+            0 => Copyright,
+            1 => IsFixedPitch,
+            2 => ItalicAngle,
+            3 => UnderlinePosition,
+            4 => UnderlineThickness,
+            5 => PaintType,
+            6 => CharstringType,
+            7 => FontMatrix,
+            8 => StrokeWidth,
+            20 => SyntheticBase,
+            21 => PostScript,
+            22 => BaseFontName,
+            23 => BaseFontBlend,
+            30 => Ros,
+            31 => CidFontVersion,
+            32 => CidFontRevision,
+            33 => CidFontType,
+            34 => CidCount,
+            35 => UidBase,
+            36 => FdArrayOffset,
+            37 => FdSelectOffset,
+            38 => FontName,
+            // Private DICT operators
+            9 => BlueScale,
+            10 => BlueShift,
+            11 => BlueFuzz,
+            12 => StemSnapH,
+            13 => StemSnapV,
+            14 => ForceBold,
+            17 => LanguageGroup,
+            18 => ExpansionFactor,
+            19 => InitialRandomSeed,
+            _ => return None,
+        })
+    }
+}
+
+/// Either a PostScript DICT operator or a (numeric) operand.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum Token {
+    Operator(Operator),
+    Operand(Number),
+}
+
+impl From<Operator> for Token {
+    fn from(value: Operator) -> Self {
+        Self::Operator(value)
+    }
+}
+
+impl<T> From<T> for Token
+where
+    T: Into<Number>,
+{
+    fn from(value: T) -> Self {
+        Self::Operand(value.into())
+    }
+}
+
+/// Given a byte slice containing DICT data, returns an iterator yielding
+/// raw operands and operators.
+///
+/// This does not perform any additional processing such as type conversion,
+/// delta decoding or blending.
+pub fn tokens(dict_data: &[u8]) -> impl Iterator<Item = Result<Token, Error>> + '_ + Clone {
```

Comment:
do we expect this to be actually used elsewhere?

Reply:
Good question. I exposed this to keep with the theme that read-fonts tries to provide low level access to encoded data.

The `entries` function is a sort of high level interface in that it consumes some information (specifically the blend vectors).

I was thinking we’d want a mid level version of this in write-fonts (where we can allocate) that can capture a more complete version of the data that can be round-tripped. This is a useful primitive for building that.